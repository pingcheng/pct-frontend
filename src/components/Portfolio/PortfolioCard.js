import React, { Component } from "react";
import PropTypes from "prop-types";

export class PortfolioCard extends Component {

    static propTypes = {
        name: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
    };

    render() {
        return (
            <div
                className="portfolio-card"
                style={{
                    backgroundImage: `url(${this.props.image})`
                }}
            >
                <div className="absolute bottom-0 p-2 text-gray-200" style={{
                    zIndex: 10
                }}>
                    <div className="text-sm font-bold">{this.props.name}</div>
                    <div className="text-xs">{this.props.description}</div>
                </div>
            </div>
        );
    }
}