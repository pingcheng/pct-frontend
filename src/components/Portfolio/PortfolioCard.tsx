import React from "react";

export type PortfolioCardProps = {
  name: string;
  description: string;
  image: string;
};

export function PortfolioCard({
  name,
  description,
  image,
}: PortfolioCardProps): JSX.Element {
  return (
    <div
      className="portfolio-card"
      role="portfolio-card"
      style={{
        backgroundImage: `url(${image})`,
      }}
    >
      <div
        className="absolute bottom-0 p-2 text-gray-200"
        style={{
          zIndex: 10,
        }}
      >
        <div className="text-sm font-bold">{name}</div>
        <div className="text-xs">{description}</div>
      </div>
    </div>
  );
}
