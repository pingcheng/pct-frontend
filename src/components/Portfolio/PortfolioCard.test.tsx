import { render, screen } from "@testing-library/react";
import { PortfolioCard } from "./PortfolioCard";

describe("test <PortfolioCard>", () => {
  const props = {
    name: "name",
    description: "description",
    image: "http://sampleUrl",
  };

  it("should render correctly", function () {
    const { container } = render(<PortfolioCard {...props} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("should render card title", function () {
    render(<PortfolioCard {...props} />);
    expect(screen.getByText(props.name)).toBeInTheDocument();
  });

  it("should render card description", function () {
    render(<PortfolioCard {...props} />);
    expect(screen.getByText(props.description)).toBeInTheDocument();
  });

  it("should render card image", function () {
    render(<PortfolioCard {...props} />);
    expect(screen.getByRole("portfolio-card")).toHaveStyle({
      backgroundImage: `url(${props.image})`,
    });
  });
});
