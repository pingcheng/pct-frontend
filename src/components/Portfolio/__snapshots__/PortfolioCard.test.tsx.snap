// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <PortfolioCard> should render correctly 1`] = `"<div class=\\"portfolio-card\\" role=\\"portfolio-card\\" style=\\"background-image: url(http://sampleUrl);\\"><div class=\\"absolute bottom-0 p-2 text-gray-200\\" style=\\"z-index: 10;\\"><div class=\\"text-sm font-bold\\">name</div><div class=\\"text-xs\\">description</div></div></div>"`;
